### PR TITLE
Create RTF with appropriate header and footer content for Pharma applications

### PR DESCRIPTION
**Pull request by rich-iannone**
This PR will add options for customizing the RTF page header and footer, allowing for custom content to be placed there and for select table content to be moved into those sections. In addition, the default font will change to 'Courier New' and a page orientation option will be made available.

Fixes: #653 
Fixes: #311

### CODE REVIEW

File: R/utils_render_rtf.R [hunk 1]
```diff
@@ -217,7 +240,7 @@ rtf_file <- function(header = NULL,
     if (length(matched_fonts) > 0) {
 
       fonttbl <-
-        rtf_fonttbl(.font_names = c("Calibri", gsub("<<FONT:|>>", "", matched_fonts)), .default_n = 1)
+        rtf_fonttbl(.font_names = c("Times", gsub("<<FONT:|>>", "", matched_fonts)), .default_n = 1)
 
       for (i in seq_along(matched_fonts)) {
         document <- gsub(matched_fonts[i], as.character(i), document, fixed = TRUE)
```
Comment (jcheng5) [inline R/utils_render_rtf.R:246]: We should replace this with gfsub at some point.

Comment (jcheng5) [reply R/utils_render_rtf.R:246]: Or `regmatches(m) <- ...`

File: R/utils_render_rtf.R [hunk 2]
```diff
@@ -230,29 +253,56 @@ rtf_file <- function(header = NULL,
...
+        paste0("\\paperh", paper_height_val),
+        "\\widowctrl\\ftnbj\\fet0\\sectd\\linex0\n",
+        "\\lndscpsxn\n",
+        paste0("\\margl", getOption("gt.rtf_page_margin_left")),
+        paste0("\\margr", getOption("gt.rtf_page_margin_right")),
+        paste0("\\margt", getOption("gt.rtf_page_margin_top")),
+        paste0("\\margb", getOption("gt.rtf_page_margin_bottom")),
+        "\n",
+        paste0("\\headery", getOption("gt.rtf_page_header_height")),
+        paste0("\\footery", getOption("gt.rtf_page_footer_height")),
```
Comment (jcheng5) [inline R/utils_render_rtf.R:274]: Move these options to `tab_options`

Comment (jcheng5) [reply R/utils_render_rtf.R:274]: Rename to non-RTF, accept arbitrary absolute units

File: R/utils_render_rtf.R [hunk 3]
```diff
@@ -1555,6 +1665,7 @@ create_footer_component_rtf <- function(data) {
 
   col_widths <-
     col_width_resolver_rtf(
+      page_orientation = dt_options_get_value(data = data, option = "page_orientation"),
```
Comment (jcheng5) [inline R/utils_render_rtf.R:1668]: This and the same table_width calculation code in create_header_component_rtf could be refactored into a common function

File: R/utils_render_rtf.R [hunk 4]
```diff
@@ -1655,3 +1766,98 @@ create_footer_component_rtf <- function(data) {
...
+    }
+
+    if (source_notes_multiline) {
+      source_notes <- paste(source_notes, collapse = "\\line ")
+    } else {
+      source_notes <- paste(source_notes, collapse = source_notes_sep)
+    }
+  } else {
+    source_notes <- ""
+  }
```
Comment (jcheng5) [inline R/utils_render_rtf.R:1854]: footnote/sourcenote generation code should be extracted into functions

Comment (rich-iannone) [reply R/utils_render_rtf.R:1854]: Added the `generate_notes_list()` function for this.

File: R/utils_render_rtf.R (hunk 2 above)
Comment (rich-iannone) [reply R/utils_render_rtf.R]: Now done.

File: R/utils_render_rtf.R [hunk 5]
```diff
@@ -199,82 +199,80 @@ rtf_header <- function(
...
-  }
+  page_orientation <- dt_options_get_value(data = data, option = "page_orientation")
+  page_width_val <- dt_options_get_value(data = data, option = paste0("page_width_", page_orientation))
+  page_height_val <- dt_options_get_value(data = data, option = paste0("page_height_", page_orientation))
+  page_margin_left_val <- dt_options_get_value(data = data, option = "page_margin_left")
+  page_margin_right_val <- dt_options_get_value(data = data, option = "page_margin_right")
+  page_margin_top_val <- dt_options_get_value(data = data, option = "page_margin_top")
+  page_margin_bottom_val <- dt_options_get_value(data = data, option = "page_margin_bottom")
+  page_header_height_val <- dt_options_get_value(data = data, option = "page_header_height")
+  page_footer_height_val <- dt_options_get_value(data = data, option = "page_footer_height")
```
Comment (jcheng5) [inline R/utils_render_rtf.R:259]: Convert all of these to twips

File: R/dt_options.R [hunk 6]
```diff
@@ -202,7 +202,17 @@ dt_options_tbl <-
...
     "row_striping_include_table_body",   FALSE,  "row",              "logical", FALSE,
     "page_orientation",                  FALSE,  "page",             "value",   "portrait",
-    "page_numbering",                    FALSE,  "page",             "value",   "none",
+    "page_numbering",                    FALSE,  "page",             "logical", FALSE,
     "page_header_use_tbl_headings",      FALSE,  "page",             "logical", FALSE,
     "page_footer_use_tbl_notes",         FALSE,  "page",             "logical", FALSE,
+    "page_height_portrait",              FALSE,  "page",             "value",   15840L,
+    "page_height_landscape",             FALSE,  "page",             "value",   12240L,
+    "page_width_portrait",               FALSE,  "page",             "value",   12240L,
+    "page_width_landscape",              FALSE,  "page",             "value",   15840L,
```
Comment (jcheng5) [inline R/dt_options.R:211]: Let's just have one page_height and page_width, and compare their sizes to determine which is the long and short sides.

File: R/utils_render_rtf.R (hunk 5 above)
Comment (rich-iannone) [reply R/utils_render_rtf.R]: Now implemented.

File: R/dt_options.R [hunk 7]
```diff
@@ -205,14 +205,12 @@ dt_options_tbl <-
...
-    "page_height_landscape",             FALSE,  "page",             "value",   12240L,
-    "page_width_portrait",               FALSE,  "page",             "value",   12240L,
-    "page_width_landscape",              FALSE,  "page",             "value",   15840L,
-    "page_margin_left",                  FALSE,  "page",             "value",   1440L,
-    "page_margin_right",                 FALSE,  "page",             "value",   1440L,
-    "page_margin_top",                   FALSE,  "page",             "value",   1440L,
-    "page_margin_bottom",                FALSE,  "page",             "value",   1440L,
-    "page_header_height",                FALSE,  "page",             "value",   720L,
-    "page_footer_height",                FALSE,  "page",             "value",   720L,
+    "page_width",                        FALSE,  "page",             "value",   8.5, # 12240 twips
```
Comment (jcheng5) [inline R/dt_options.R:208]: I thought these sizes were going to be literally `"8.5in"`?

Comment (rich-iannone) [reply R/dt_options.R]: Yes, thanks for catching this. Makes infinitely more sense and we did talk about it.

**Review by jcheng5 — APPROVED** [review]

**Pull request merged**